### ISSUE TITLE
Can't build from source on arm32v7

### ISSUE THREAD

**Issue by aptalca** [maxbachmann/Levenshtein#36]
Hi, I'm trying to install `Levenshtein 0.20.4` on Ubuntu Focal arm32v7, which builds from source as there are no wheels available, and getting the following error:
https://hastebin.com/opewiyapoq.sql

I did install `cmake` (os package), and `ninja` and `rapidfuzz` pip packages prior to `pip install Levenshtein`

Thanks

**Comment by maxbachmann** [comment]
This error is caused by the cmake version. It appears that `WITH_SOABI` was added in cmake 3.17, but you have cmake 3.16 installed. Note that this likely affects the `rapidfuzz` package as well. I assume it simply installed the pure Python fallback, since the build failed.

@henryiii cmake 3.17 was released in 2020, so I think it is a bit early to require it.

**Comment by henryiii** [comment]
Ideally we'll provide a backport of FindPython so it will be easy to support CMake 3.15+. We can't go older than that in scikit-build-core because it uses the FileAPI which was introduced in 3.15, as well as a host of other 3.14 & 3.15 features.

In this case, WITH_SOABI was only really required for PyPy, since CPython will load "thing.so" also; I think PyPy doesn't because it doesn't want to accidentally load a CPython .so.

**Comment by maxbachmann** [comment]
Yes I think I should be able to manually set the name of the output file to the appropriate version for older versions of cmake

**Comment by henryiii** [comment]
You don't have to except for PyPy, the non-SOABI name works too. But you can (pybind11 does by default, if you'd like to see what was done there).

**Comment by maxbachmann** [comment]
I guess if I do it for PyPy, I have to write the code anyways, so it does not really matter at that point

**Comment by henryiii** [comment]
Does PyPy support arm32v7? Their alternative arch support usually is limited.

**Comment by maxbachmann** [comment]
I do not know, but looking at the cmake implementation `WITH_SOABI` is simply implemented as:
```
if (PYTHON_ADD_LIBRARY_WITH_SOABI AND ${prefix}_SOABI)
  get_property (suffix TARGET ${name} PROPERTY SUFFIX)
    if (NOT suffix)
      set (suffix "${CMAKE_SHARED_MODULE_SUFFIX}")
    endif()
    set_property (TARGET ${name} PROPERTY SUFFIX ".${${prefix}_SOABI}${suffix}")
endif()
```
which is easy enough to backport

**Comment by henryiii** [comment]
Yes, that's about what pybind11 does too.

**Comment by maxbachmann** [comment]
@aptalca this should be fixed in `v0.20.5`. Can you confirm this?

**Comment by aptalca** [comment]
@maxbachmann yup, I can confirm `0.20.5` builds fine from source on arm32v7 with the os package cmake 3.16

Thanks so much for the quick fix
